server/static/editor-plugins/addons.js

```diff
@@ -1,10 +1,99 @@
 // https://h5p.org/adding-text-editor-buttons
 
+function getFixMathTypeDialog() {
+  let offsetTop = 0;
+  let viewportHeight = 0;
+  let parentOffsetTop = 0;
+  let parentWidth = 0;
+  let parentHeight = 0;
+  let mathTypeInstances = [];
+  let waitingForUpdate = false;
+
+  const setMathTypePosition = () => {
+    const visibleInstances = mathTypeInstances.filter(
+      (editor) => !editor.classList.contains('wrs_closed')
+    );
+    if (waitingForUpdate || visibleInstances.length === 0) return;
+    waitingForUpdate = true;
+    requestAnimationFrame(() => {
```

Comment:
Maybe `waitingForUpdate` and `shouldUpdate` could be combined into one variable named something like `isUpdatePending`? It seems like the intent is to update once after mutations have occurred.

If so, then it seems `requestAnimationFrame` only needs to run if it's not already queued up

Reply:
~Part of me wants to keep the observer as removed from `setMathTypePosition` as possible because I think `setMathTypePosition` could be repurposed into a more generic `update` function that could deal with positioning a variety of dialogs. Hopefully we will not need to position more elements, but I like to keep options open.~

~The `shouldUpdate` in the observer is being used to determine if there is a reason to call `setMathTypePosition`. It's a constant because I wanted to avoid nesting it inside the `if`.~

This comment convinced me that I should finish the "evergreen" approach instead of making something singularly aimed at the mathtype editor window. Granted, I am not going to add more complex tiling logic yet because I am not sure it will be required.